opentracing_instrumentation/client_hooks/sqlalchemy.py

```diff
@@ -64,9 +64,13 @@ def before_cursor_execute(conn, cursor, statement, parameters, context,
                                    statement.split(' ', 1)[0].upper())
         span = utils.start_child_span(
             operation_name=operation, parent=current_span_func())
-        span.set_tag(ext_tags.SPAN_KIND, ext_tags.SPAN_KIND_RPC_CLIENT)
+        span.set_tag(tags.SPAN_KIND, tags.SPAN_KIND_RPC_CLIENT)
+        span.set_tag(tags.COMPONENT, 'sqlalchemy')
+        if conn.engine:
+            span.set_tag(tags.DATABASE_TYPE, conn.engine.name)
+            span.set_tag(tags.DATABASE_INSTANCE, repr(conn.engine.url))
```

Comment:
could't this accidentally include the password?

Reply:
`repr` on a SQLAlchemy URL object calls its `__to_string__`, with its default `hide_password=True`. I can change this to explicitly call `__to_string__(hide_password=True)`, if you think that'd be safer.

https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/engine/url.py#L77-L109

Reply:
just looked up the definition of `repr`:

> `repr()` prints “official” representation of an object (means using the “official” string representation we can reconstruct the object).

By that definition, SQLAlchemy is doing the wrong thing since the URL is not reconstructable if the password is dropped. Which might mean they could change it in the future.

On the other hand, calling a private method `__to_string__` seems even worse.

Reply:
Yeah, the defintion of `repr()` from the [Python documentation](https://docs.python.org/2/library/functions.html#repr) seems to allow for some flexibility:

> Return a string containing a **printable representation** of an object. This is the same value yielded by conversions (reverse quotes). It is sometimes useful to be able to access this operation as an ordinary function. For **many** types, this function **makes an attempt** to return a string that would yield an object with the same value when passed to eval(), otherwise the representation is a string enclosed in angle brackets that contains the name of the type of the object together with additional information often including the name and address of the object. A class can control what this function returns for its instances by defining a __repr__() method.

Emphasis mine. My thoughts:
- "printable representation" makes sense for a string with redacted, sensitive information that we might want to "print" (in this case, to tracing)
- "many" types, "makes an attempt" seem to imply that it's not an absolute requirement that `__repr__` be able to be passed to `eval()` to reconstruct an object.

I could go either way. I think the use in this context makes sense, but I also think calling `__to_string__` would probably reasonable as it doesn't seem *private* as much as a weirdly-named magic method.

Reply:
What if we create a unit test for that? Pass a string w/ password and make sure that the tag doesn't contain the password string.

Reply:
Sure. I added such a test; let me know what you think.